### PR TITLE
add CrudMethode for Slot in facade

### PR DESCRIPTION
**Pull request by Theo-Hafsaoui**
- [x]  code
- [x]  test

ps: sorry for being late, some unexpected holiday festivity

### CODE REVIEW

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 1]
```diff
@@ -151,6 +212,48 @@ public static List<Slot> getSchedule(Group group, Interval timerange) {
...
+     * @param slot
+     * @return
+     */
+    public static float getPercentageOf(Group group, Slot slot) {
+        int total = 1;
+        int passed = 0;
+        List<Slot> listOfSlot;
+        SlotDAO dao = SlotDAO.getInstance();
+        try {
+            listOfSlot = dao.getSlotOfGroup(group);
```
Comment (Desmondmies) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:244]: why using a group, shouldn't it be list of subject ?

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 2]
```diff
@@ -151,6 +212,48 @@ public static List<Slot> getSchedule(Group group, Interval timerange) {
...
+        } catch (SQLException e) {
+            return -1;
+        }
+        for (Slot xslot :
+                listOfSlot) {
+            if (xslot.getTimeRange().isBefore(slot.getTimeRange())) {
+                passed += 1;
```
Comment (Desmondmies) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:251]: Change `passed += 1`
to
`passed += xslot.getTimeRange().toDuration().toHours()`

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 3]
```diff
@@ -151,6 +212,48 @@ public static List<Slot> getSchedule(Group group, Interval timerange) {
...
+            }
+            total += 1;
```
Comment (Desmondmies) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:253]: `total += 1`
Can be removed entirely, use 
`total = slot.getSubject().getHourCountMax()`

Doing those two changes should give you the estimate percentage of passed subject over subject max hours

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java (hunk 1 above)
Comment (Theo-Hafsaoui) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:244]: Why ? You want the rest of the slot for a subject for a group we might have several group with the same subject(physique in L1 for instance)

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java (hunk 2 above)
Comment (Theo-Hafsaoui) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:251]: Yeap didn't thought of that

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java (hunk 1 above)
Comment (Desmondmies) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:244]: We don't want to evaluate the number of hours for many groups, only for the one group the connected student is in.
So in this case list of groups is not relevant, using only one group is what we need

Comment (Theo-Hafsaoui) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:244]: but I don't use a listOfGroup but just the group of the student

**Review by Desmondmies — APPROVED** [review]

**Review by anastygnome — APPROVED** [review]
lgtm, hope you enjoyed your holiday and happy new year ;)

**Pull request merged**